lib/widgets/calendar_dialog.dart

```diff
@@ -0,0 +1,188 @@
+          onPressed: _clearFilter,
+        ),
+        new FlatButton(
+          child: new Text("Apply Filter", style: theme.buttonStyle),
+          onPressed: () {
+            if (_startDate != widget._firstDate || _endDate.day != new DateTime.now().day) {
+              DateTime startDate = new DateTime(_startDate.year, _startDate.month, _startDate.day, 0, 0, 0);
+              DateTime endDate = new DateTime(_endDate.year, _endDate.month, _endDate.day, 0, 0, 0);
+              if(startDate.compareTo(endDate) == 0) {
+                endDate = new DateTime(_endDate.year, _endDate.month, _endDate.day, 23, 59, 59, 999);
+                Navigator.of(context).pop([startDate, endDate]);
+              } else {
+                Navigator.of(context).pop([_startDate, _endDate]);
+              }
+            } else {
+              Navigator.of(context).pop([null,null]);
+            }
+          },
+        ),
+      ],
+    );
+  }
+
+  Future<Null> _selectStartDate(BuildContext context) async {
+    DateTime selectedStartDate = await showBreezDatePicker(
+      context: context,
+      initialDate: _startDate,
+      firstDate: widget._firstDate,
+      lastDate: DateTime.now(),
+    );
+    Duration difference = selectedStartDate.difference(_endDate);
+    if (selectedStartDate != null && difference.inDays < 0) {
+      setState(() {
+        _startDate = selectedStartDate;
+        _startDateController.text = _formatTransactionDate(_startDate);
+        _endDateController.text = _formatTransactionDate(_endDate);
+      });
+    } else {
+      setState(() {
+        _startDate = selectedStartDate;
+        _endDate = DateTime.now();
+        _startDateController.text = _formatTransactionDate(_startDate);
+        _endDateController.text = _formatTransactionDate(_endDate);
+      });
+    }
+  }
+
+  Future<Null> _selectEndDate(BuildContext context) async {
+    DateTime selectedEndDate = await showBreezDatePicker(
+      context: context,
+      initialDate: _endDate,
+      firstDate: widget._firstDate,
+      lastDate: DateTime.now(),
+    );
+    Duration difference = selectedEndDate.difference(_startDate);
+    if (selectedEndDate != null && difference.inDays > 0) {
+      setState(() {
+        _endDate = selectedEndDate;
+        _startDateController.text = _formatTransactionDate(_startDate);
+        _endDateController.text = _formatTransactionDate(_endDate);
+      });
+    } else {
+      setState(() {
+        _startDate = widget._firstDate;
+        _endDate = selectedEndDate;
+        _startDateController.text = _formatTransactionDate(_startDate);
+        _endDateController.text = _formatTransactionDate(_endDate);
+      });
+    }
+  }
+
+  _clearFilter() {
+    setState(() {
+      _startDate = widget._firstDate;
+      _endDate = new DateTime.now();
+      _startDateController.text = "";
+      _endDateController.text = "";
+    });
+  }
+
+  String _formatTransactionDate(DateTime date) {
```

Comment:
_formatTransactionDate is not relevant here, there is no transactions...
I addition to that it is duplicated there are lots of uses of: 
* DateFormat.yMd 
* DateFormat.Md
This means we need a simple DateUtil that have two functions:
1. formatMonthDate
2. formatYearMonthDay
And use it whenever needs formatting.